### DISCUSSION TITLE
Which api does not support win7?

### DISCUSSION THREAD

**Discussion by wj8400684** [amwx/FluentAvalonia#300] [Q&A]
Which api does not support win7?

**Comment by timunie** [comment]
Can you clarify your question a bit more? Win7 isn't supported by M$, so it's not (officially) supported by Avalonia. But until today, I had no issue running on Windows 7. I don't know it that may change in upcomming versions.

**Comment by fitdev** [comment]
I second this question in regards to FluentAvalonia. I was quite disappointed to see that it does not support Win 7, 8, and 8.1. I tried building and running the samples app myself. Works great on Windows 10 (no surprises there), but does not work at all on Windows 7 - simply crashes. Yet, "Vanilla" Avalonia works and runs just fine on Windows 7 when built against DotNet 7. I tweaked the code on my end a bit and removed WinRT dependence, but it still crashed on Windows 7. So I am wondering which specific APIs make it crash on Windows 7?

**Comment by amwx** [comment]
- `FluentAvaloniaTheme` uses WinRT apis for system theme detection.
- `AppWindow` uses some WINAPI calls that are only available in Win 8/10+ (like `AdjustWindowRectExForDPI` is only Win10+)
- One or two other controls may use a couple WINAPI calls for system metrics, but I think those are not limited to 10+

You \*might\* be able to get an app running, so long as:
- You keep PreferSystemTheme and PreferUserAccentColor are left at `false` (technically the WinRT stuff is wrapped in try/catch so if it fails it doesn't stop the app, it just falls back). I think MicroCom can handle that (since Avalonia has MicroCom stuff in it too and it works fine)
- You don't use AppWindow

Both of these are extensively used in the Sample app.

I have no computers running Win 7/8/8.1 to test anything (or diagnose if an issue came up) and as they're legacy out-of-support operating systems now, I'm not going to invest extra time to ensure FA stays compatible. So its more of a warning rather than a hard no. If you can get it working, great, but you're on your own.

**Reply by fitdev** [reply]
Thanks so much for a detailed response and pointing me in the right direction. And thank you for this great library! I suspect it's the AppWindow then which needs work. I will try to see if I can make some of its features conditional based on the OS version.
